typewise_alert.py

```diff
@@ -30,8 +30,15 @@ def validate_input(coolingType, temperatureInC):
     
 def check_and_alert(alertTarget, batteryChar, temperatureInC):
     breachType = classify_temperature_breach(batteryChar['coolingType'], temperatureInC)
-    if not breachType == "NORMAL":
-       return(alertTarget_type[alertTarget](breachType))     
+    if not breachType == "NORMAL" and validate_alertTarget(alertTarget, breachType):
+       return(alertTarget_type[alertTarget](breachType))
```

Comment:
If the program emails instead of sending to controller (due to a bug), your tests will still pass.
Strengthen the tests to catch this.